src/snap/NotificationsEmitter.ts

```diff
@@ -0,0 +1,86 @@
+import { StrictEventEmitter } from "strict-event-emitter";
+import { DappeteerPage } from "../page";
+import * as dappeteer from "../../src";
+import { clickOnElement, profileDropdownClick } from "../helpers";
+import { NotificationItem, NotificationList } from "./types";
+
+interface EventsMap {
+  notification: (notification: NotificationItem) => void;
+}
+
+class NotificationsEmitter extends StrictEventEmitter<EventsMap> {
+  private notificationsTab: DappeteerPage;
+  private readonly page: DappeteerPage;
+
+  constructor(
+    protected metamask: dappeteer.Dappeteer,
+    private notificationTimeout: number = 30000
+  ) {
+    super();
+    this.page = metamask.page;
+  }
+
+  private async exposeEmitNotificationToWindow(): Promise<void> {
+    await this.notificationsTab.exposeFunction(
+      "emitNotification",
+      (notification: NotificationItem) => {
+        this.emit("notification", notification);
+      }
+    );
+  }
+
+  private async openNotificationPage(): Promise<void> {
+    await this.page.bringToFront();
+    await profileDropdownClick(this.page);
+    await clickOnElement(this.page, "Notifications");
+
+    const newPage = await this.page.browser().newPage();
+    await newPage.goto(this.page.url());
+
+    await newPage.waitForSelector(".notifications__container", {
+      timeout: this.notificationTimeout,
+    });
+    this.notificationsTab = newPage;
+  }
```

Comment:
think it is better first to open a new page and then navigate to notifications tabs from it, that moving there then open a new page and navigate back to "home"